projects/packages/blaze/src/class-dashboard-rest-controller.php

```diff
@@ -386,6 +391,10 @@ public function get_dsp_blaze_posts( $req ) {
 			return array();
 		}
 
+		if ( ! $this->is_posts_synced() ) {
```

Comment:
@sbarbosa I'm not sure if it is needed for `get_dsp_blaze_posts()` method as well.

Reply:
Yes, I think we also need to check on that other endpoint.

The `get_dsp_blaze_posts` is called when a user clicks the Promote button on the header of the page. Because we don't have a Product ID at that point, we show a full list of post for the user to choose. 

We internally call the same WPCOM endpoint, so it will also have the same sync problem as the other one.
This is the page I was talking about:
![Screenshot 2024-06-14 at 1 20 27 PM](https://github.com/Automattic/jetpack/assets/1258162/8e1c0ea7-7db5-4313-b1bc-e2fbdc603b6a)